### PR TITLE
feat(detector-aws): add sync version for all detectors

### PR DESCRIPTION
**Pull request by david-luna**
## Which problem is this PR solving?

As suggested by @blumamir this PR aggregates the previous PRs to add sync detectors in the `@opentelemetry/resource-detector-aws` package. Previous PRs will be closed

## Short description of the changes

- add detectors for Beanstalk, EC2, ECS, EKS & Lambda complying with the `DetectorSync` interface.
- add tests
- make use of named exports

### CODE REVIEW

File: detectors/node/opentelemetry-resource-detector-aws/test/detectors/AwsBeanstalkDetector.test.ts [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 import * as assert from 'assert';
 import * as sinon from 'sinon';
-import { awsBeanstalkDetector, AwsBeanstalkDetector } from '../../src';
+import { awsBeanstalkDetector, AwsBeanstalkDetectorSync } from '../../src';
```
Comment (david-luna) [inline detectors/node/opentelemetry-resource-detector-aws/test/detectors/AwsBeanstalkDetector.test.ts:19]: note for reviewer: `awsBeanstalkDetector` is delegating the job to `awsBeanstalkDetectorSync` so we need to apply the stubs into `AwsBeanstalkDetectorSync` class

File: detectors/node/opentelemetry-resource-detector-aws/test/detectors/AwsBeanstalkDetector.test.ts [hunk 2]
```diff
@@ -105,13 +108,14 @@ describe('BeanstalkResourceDetector', () => {
...
-      .stub(AwsBeanstalkDetector, 'fileAccessAsync' as any)
+      .stub(AwsBeanstalkDetectorSync, 'fileAccessAsync' as any)
       .rejects(err);
     readStub = sinon
-      .stub(AwsBeanstalkDetector, 'readFileAsync' as any)
+      .stub(AwsBeanstalkDetectorSync, 'readFileAsync' as any)
       .resolves(JSON.stringify(data));
 
     const resource = await awsBeanstalkDetector.detect();
+    await resource.waitForAsyncAttributes?.();
```
Comment (david-luna) [inline detectors/node/opentelemetry-resource-detector-aws/test/detectors/AwsBeanstalkDetector.test.ts:118]: note for reviewer: now `awsBeanstalkDetector.detect()` resolves to a promise before the async attributes are resolved so we need to await

**Review by blumamir — APPROVED** [review]
LGTM.

Thanks for merging all the package detectors into one PR 🙏

**Comment by david-luna** [comment]
Error in `hapi` tests on node 18 due to a timeout :(

```
26 passing (2s)
  1 failing

  1) Hapi Instrumentation - Core Tests
       ESM
         should work with ESM modules:
     Error: Timeout of 2000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/home/runner/work/opentelemetry-js-contrib/opentelemetry-js-contrib/plugins/node/opentelemetry-instrumentation-hapi/test/hapi.test.ts)
      at listOnTimeout (node:internal/timers:569:17)
      at processTimers (node:internal/timers:512:7)
```

**Pull request merged**